typing/typecore.ml

```diff
@@ -2811,8 +2856,28 @@ and type_pat_aux
       let alloc_mode =
         cross_left !!penv expected_ty alloc_mode.mode
       in
+      let mode, kind =
+        match mutable_flag with
+        | Immutable -> alloc_mode, Val_reg
+        | Mutable ->
+            let m0 = Value.Comonadic.newvar () in
+            let mode = mutvar_mode ~loc ~env:!!penv m0 alloc_mode in
+            let kind =
+              Val_mut
+              (* CR-someday let_mutable: move the sort calculation elsewhere *)
```

Comment:
I think we should probably delete this comment - if we left a comment for every small thing we wish there were another way to do, the typechecker would be 90% comments.

It's not crazy to leave it, but I'd suggest then making it not a CR, just a comment, and being more specific: the thing that is a little sad is that we have to check all pattern variables are representable, so we're already calculating the sort of this elsewhere, but it's too far removed to easily pass that information here.